guidelines/coding-guidelines.md

```diff
@@ -11,3 +11,59 @@ For public facing interfaces we favor read-only collections and enumerables. For
 * Avoid using `System.Linq`
 * Avoid using `foreach` over collections that do not have a struct enumerator
 
+## Dependencies
+
+### 3rd party dependencies
+
+#### Package dependencies
+
+We want to compile against the lowest major to catch breaking changes
+
+PackageReference: `[Major.0.0, Major+1)`
```

Comment:
True, until such a time that the downstream needs to take advantage of an API introduced in a minor, in which case the dependency would become `[Major.Minor.0, Major+1)`

Reply:
Is the text going to be altered to reflect this? How about

> We want to compile against the lowest compatible version to catch breaking changes
> 
> PackageReference: `[Major.Minor.0, Major+1)`

Reply:
@adamralph I've updated to text now , see https://github.com/Particular/NServiceBus/pull/4930/commits/23ab77e8bb40d1e14a537a27be69f2b2612fbfe2